backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -367,48 +289,11 @@ public Response deleteEndpoint(@Context SecurityContext sec, @PathParam("id") UU
         }
         checkSystemEndpoint(endpointType);
 
-        Endpoint ep = endpointRepository.getEndpoint(orgId, id);
-        if (isForOpenBridge(ep)) {
-            /*
-             * Before the endpoint and its processor get deleted, we need to
-             * check that the processor was in an actionable state before
-             * updating it in RHOSE.
-             *
-             * This will achieve two goals:
-             *
-             * 1. Avoid returning the RHOSE error directly to the
-             * user.
-             * 2. Avoid unnecessarily calling RHOSE.
-             *
-             * Check https://issues.redhat.com/browse/RHCLOUD-22355
-             * for more information.
-             */
-            if (ep.getStatus() != EndpointStatus.READY && ep.getStatus() != EndpointStatus.FAILED) {
-                throw new BadRequestException(
-                    "the processor associated with the endpoint is not ready to be deleted"
-                );
-            }
-
-            try {
-                ep.setStatus(EndpointStatus.DELETING);
-                CamelProperties properties = ep.getProperties(CamelProperties.class);
-                String processorId = properties.getExtras().get(OB_PROCESSOR_ID);
-                try {
-                    bridgeApiService.deleteProcessor(bridge.getId(), processorId, bridgeAuth.getToken());
-                } catch (WebApplicationException e) {
-                    String path = "DELETE " + BASE_PATH + "/{bridgeId}/processors/{processorId}";
-                    rhoseErrorMetricsRecorder.record(path, e);
-                    throw e;
-                }
-            } catch (Exception e) {
-                Log.warn("RHOSE processor deletion failed", e);
-            }
-        } else {
-            endpointRepository.deleteEndpoint(orgId, id);
-        }
+        endpointRepository.deleteEndpoint(orgId, id);
 
         // Clean up the secrets in Sources.
         if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
+            Endpoint ep = endpointRepository.getEndpoint(orgId, id);
```

Comment:
I think the `deleteEndpoint` should go after the Sources references removal, because otherwise the `getEndpoint` function will throw a Not Found, no?

Reply:
I think so, yes. The order hasn't been changed by this PR though so I'm surprised we didn't see this before while testing.

Reply:
Oops, I can open a new PR after we merge this one, then